katim/KATH5toAIPS.py

```diff
@@ -646,6 +669,9 @@ def ConvertKATData(outUV, katdata, meta, err, static=None, blmask=1.e10, stop_w=
             if doweight==False:
                 wt[:] = 1.
             vs = scan_vs[:nint]
+            if antphasedict_rad is not None:
+                for iprod,(input0,input1) in enumerate(katdata.corr_products):
+                    vs[:,:,iprod]*=np.exp(1j*(antphasedict_rad[input0][katdata.channels]-antphasedict_rad[input1][katdata.channels]))[np.newaxis,:]
```

Comment:
Maybe I'm missing something, but couldn't you have the function above that generates error message also generate the multiplier (` np.exp(1j*(antphasedict_rad[input0][katdata.channels]-antphasedict_rad[input1][katdata.channels]))[np.newaxis,:]` ) so that it can broadcast into vs. This would speed up the calculation a lot.

Reply:
Actually, thinking about it some more maybe not since `nchan,ncorr` might be too much memory in general.